### ISSUE TITLE
On Linux, clipboard items in the main UI are massive

### ISSUE THREAD

**Issue by thennothinghappened** [hluk/CopyQ#2093]
**Describe the bug**
Clipboard items take up more than the entire screen and make it nearly impossible to use

**To Reproduce**
1. Open CopyQ
2. Copy some text
3. Left Click the tray icon to open the UI

**Expected behavior**
Items should be a reasonable size

**Screenshots**
![image](https://user-images.githubusercontent.com/35243139/190951636-0cd1b2bd-a855-492c-993a-b249bd9a4b74.png)

**Version, OS and Environment**
```
CopyQ Clipboard Manager 6.2.0
Qt: 5.15.5
KNotifications: 5.97.0
Compiler: GCC
Arch: x86_64-little_endian-lp64
OS: Arch Linux
```
Running on i3, nvidia proprietary driver.

**Additional context**
I tried using `QT_AUTO_SCREEN_SCALE_FACTOR=0` as I had success using that to fix another QT app, but it doesn't work here.

**Comment by hluk** [comment]
Can you try the latest version (6.3.0)?

**Comment by nassets** [comment]
我的 Windows 11 今天更新了 6.3.0 版本，出现了这种情况
<img width="448" alt="S0919_949" src="https://user-images.githubusercontent.com/81686813/191000078-0a2f1b6c-34e1-4d5e-bc73-f710875e123f.png">

**Comment by richlux** [comment]
I have the same problem on Windows 10. It just shows a big blank area and you can't see the actual copied text

**Comment by hluk** [comment]
Can you test it with this [development build](https://ci.appveyor.com/project/hluk/copyq/builds/44817892/artifacts)?

This has been problem on some Linux platforms for some time (#1728). It looks like recent improvements broke it also on Windows.

**Comment by richlux** [comment]
Same problem, but it looks like it's only a problem if "Show Numbers" is turned off.  If they are turned on, it seems to work.  Is there a reason why you removed some of the space between the line number and the copied text?  That's why I turned numbers off because it was hard to read with them on.

Rich

**Comment by hluk** [comment]
bq. Is there a reason why you removed some of the space between the line number and the copied text?

Can you share screenshot? There should be still some space between the number and item text.

**Comment by richlux** [comment]
There's still some space, it's just less than before.  I can probably get used to it after a while.

![image](https://user-images.githubusercontent.com/25157703/191047359-bfbd4fd4-0c2b-4138-94cf-ced6e6661655.png)

**Comment by FreeHarry** [comment]
> Same problem, but it looks like it's only a problem if "Show Numbers" is turned off. If they are turned on, it seems to work. Is there a reason why you removed some of the space between the line number and the copied text? That's why I turned numbers off because it was hard to read with them on.
> 
> Rich

I have the same problem on Windows 10, regardless if "Show Numbers" is turned ON or OFF.

**Comment by in-in** [comment]
same
![22262190348](https://user-images.githubusercontent.com/8797432/191062203-4d83d488-7ae4-4512-a88e-86a89b15bded.png)


copyq: 6.3.0
OS: Linux Mint 20.2 Cinnamon

**Comment by hluk** [comment]
I've updated the Windows build please check again: https://ci.appveyor.com/project/hluk/copyq/builds/44822264/artifacts

For Linux see: https://github.com/hluk/CopyQ/issues/1728#issuecomment-1251055850

I will fix the row number margin later.

**Comment by richlux** [comment]
I still see the same problem when Show Numbers is off, but when it is on, it seems to be working, except....  When I open CopyQ after I've copied something new, I can see CopyQ display the big boxes for the new items and then refresh to display the list correctly.  It happens very quickly, but it's still noticeable.  Did something change with the way CopyQ displays the list?  I'm also seeing issues when I have Anti-aliasing" set.  When I have text that contains tabs to indent lines, CopyQ displays them in the list, but when I click on them, the item refreshes and the indentation changes.  If I close CopyQ and reopen it, the display of the items resets until I click on them again.  I posted a video of the issue below.

https://user-images.githubusercontent.com/25157703/191115239-5bc3b313-c1bf-490a-bd2d-bfaa3be11621.mp4

**Comment by hluk** [comment]
Yet another dev build: https://ci.appveyor.com/project/hluk/copyq/builds/44826292/artifacts

Please test if possible. I've tested on my side extensively but I currently cannot verify it on Windows.

The build **also fixes the space between row numbers and items** and it is configurable via Edit Theme button in Appearance config tab - the option name is `num_margin`.

**Comment by brkerez** [comment]
> Yet another dev build: [ci.appveyor.com/project/hluk/copyq/builds/44826292/artifacts](https://ci.appveyor.com/project/hluk/copyq/builds/44826292/artifacts)
> 
> Please test if possible. I've tested on my side extensively but I currently cannot verify it on Windows.
> 
> The build **also fixes the space between row numbers and items** and it is configurable via Edit Theme button in Appearance config tab - the option name is `num_margin`.

I think on windows the problem with no space between numbers and text is gone ;)
**before**
![image](https://user-images.githubusercontent.com/313735/191216803-121f311d-2716-4505-b399-650a54d93f13.png)
**after**
![image](https://user-images.githubusercontent.com/313735/191216832-fe61f0e1-c6e9-4b22-9138-cdda0065d6a8.png)

**Comment by FreeHarry** [comment]
> Yet another dev build: https://ci.appveyor.com/project/hluk/copyq/builds/44826292/artifacts
> 
> Please test if possible. I've tested on my side extensively but I currently cannot verify it on Windows.
> 
> The build **also fixes the space between row numbers and items** and it is configurable via Edit Theme button in Appearance config tab - the option name is `num_margin`.

Have the same problems as before with this new version. If I scroll up and down in the list the items are refreshed and will be shown correctly. But initially not (when open the CopyQ window).

**Comment by hluk** [comment]
@FreeHarry Do you also mean there is no space between number and items?

After starting the dev build, can you check that the version (in About dialog - Shift+F1) is "6.3.0-5-g098a65f0"?

**Comment by FreeHarry** [comment]
> @FreeHarry Do you also mean there is no space between number and items?
> 
> After starting the dev build, can you check that the version (in About dialog - Shift+F1) is "6.3.0-5-g098a65f0"?

When opening CopyQ it looks so ...

![2022-09-20_15-08-45--copyq](https://user-images.githubusercontent.com/24735031/191266752-c41568b9-6004-4403-a824-0403d19852c3.png)

After scrolling a little bit, items are shown ...

![2022-09-20_15-09-31--copyq](https://user-images.githubusercontent.com/24735031/191266946-3467a7bf-0327-4aff-a6f7-4cf7987c6c51.png)

And here is the version I used ....

![2022-09-20_15-10-16--copyq](https://user-images.githubusercontent.com/24735031/191266987-05d4d3b6-3a70-4bbe-8ba3-b6fcc1f1a069.png)

**Comment by hluk** [comment]
Thanks for testing!

I've added one more refresh when the list shows up in this new build: https://ci.appveyor.com/project/hluk/copyq/builds/44830512/artifacts

Please re-test when possible.

**Comment by FreeHarry** [comment]
> Thanks for testing!
> 
> I've added one more refresh when the list shows up in this new build: https://ci.appveyor.com/project/hluk/copyq/builds/44830512/artifacts
> 
> Please re-test when possible.

![2022-09-20_16-30-15--copyq](https://user-images.githubusercontent.com/24735031/191285677-e4803197-424a-4212-bd81-58a0b1db83b0.png)

Unfortunately it is the same behavior as before.

**Comment by richlux** [comment]
Same behavior for me too. It seems to be working with Show Numbers checked, but when not checked, I get big blank boxes until I scroll in the list.  Also, even when it works with Show Numbers checked, the refresh is distracting.  Whenever I open the CopyQ window all the items I copied since the last time I opened CopyQ show as big empty boxes and then quickly refresh to show the correct info.  Older versions never did this, so something definitely changed.

Rich

**Comment by hluk** [comment]
How about this build: https://ci.appveyor.com/project/hluk/copyq/builds/44831930/artifacts

The only difference is that the new build uses a postponed update.

Otherwise I'm out of ideas. I could not find a bug in code where the difference in Show Numbers option matters.

**Comment by richlux** [comment]
I still see the same behavior with this build. I'm not concerned with the issue when "Show Numbers" is not selected because I use Show Numbers, but the fact that every time CopyQ is opened all the new items are big and then refresh smaller is quite annoying. I just installed 6.2 and it does not have the same issue.  When CopyQ is opened the new items aren't big and empty before the refresh.

Rich

**Comment by thennothinghappened** [comment]
> Can you try the latest version (6.3.0)?

heya!
sorry about the wait on the reply, but I've just tried out 6.3.0 and it works correctly now, thanks!!
I'm not closing the issue since from the replies there's some other problem going on on windows.

**Comment by FreeHarry** [comment]
> How about this build: https://ci.appveyor.com/project/hluk/copyq/builds/44831930/artifacts
> 
> The only difference is that the new build uses a postponed update.
> 
> Otherwise I'm out of ideas. I could not find a bug in code where the difference in Show Numbers option matters.

With option "Scrollbars" is ON, updating on Windows 10 is correct in all aspects.

Otherwise I got this big empty boxes when reopening the window. But only for new added items. If I did no copy operation, existing items are shown well every time.

**Comment by hluk** [comment]
I've completely rewritten the rendering code. Please help testing it.

Here is a test build for Windows: https://ci.appveyor.com/project/hluk/copyq/builds/44855887/artifacts

For Linux, there is a Flatpak image you can install using the latest instructions from flathubbot: https://github.com/flathub/com.github.hluk.copyq/pull/44#issuecomment-1255259412

**Comment by hluk** [comment]
I'm going to close this in favor of: #1728

Please follow there.
